go/client/cmd_config.go

```diff
@@ -103,11 +103,25 @@ func (v *CmdConfigSet) ParseArgv(ctx *cli.Context) error {
 			return fmt.Errorf("Missing string value argument")
 		}
 		s := args[1]
+		if !ctx.IsSet("string") && v.looksLikeBool(s) {
```

Comment:
Since string is default to true, ctx.Bool("string") returns true when no flag is present.  IsSet returns true only if the flag was actually present.